### PR TITLE
Loki: Fix error when changing operations with different parameters

### LINKED ISSUE #51657: Loki: Error when changing from `unwrap` to `binary operation`
**What happened**:
Loki Query Builder crashed when changing from `unwrap` to `binary operation`

**What you expected to happen**:
Query builder should not crash.

**How to reproduce it (as minimally and precisely as possible)**:

https://user-images.githubusercontent.com/8092184/176782809-30cc7ee4-2fb4-4b5b-9879-24a88b04ebb4.mov

### PR DESCRIPTION
**Pull request by svennergr**
**What this PR does / why we need it**:

Currently the Loki/Prometheus Query Builder crashes when changing from an operation with a parameter to an operation without parameters. The change in this PR copies the operation's `defaultParams` and overwrites any set parameters.

**Which issue(s) this PR fixes**:

Fixes #51657

**Special notes for your reviewer**:

https://user-images.githubusercontent.com/8092184/177498492-78f910a4-f191-479f-83c9-68629e0185a3.mov

### CODE REVIEW

**Review by ivanahuckova — APPROVED** [review]
Nice! LGTM.

File: public/app/plugins/datasource/prometheus/querybuilder/shared/OperationHeader.tsx [hunk 1]
```diff
@@ -79,7 +79,14 @@ export const OperationHeader = React.memo<Props>(
                 if (value.value) {
                   // Operation should exist if it is selectable
                   const newDef = queryModeller.getOperationDef(value.value.id)!;
-                  let changedOp = { ...operation, id: value.value.id };
+
+                  // copy default params, and override with all current params
+                  const newParams = [...newDef.defaultParams];
+                  for (let i = 0; i < Math.min(operation.params.length, newParams.length); i++) {
+                    newParams[i] = operation.params[i];
```
Comment (aocenas) [inline public/app/plugins/datasource/prometheus/querybuilder/shared/OperationHeader.tsx:86]: should we check param names or something? Or expected value? What will happen if in one op param is label but in second it should be a number value?

Comment (svennergr) [reply public/app/plugins/datasource/prometheus/querybuilder/shared/OperationHeader.tsx:86]: We should definitely improve the handling here. I.e. changing a operation to `Binary operation with query` does currently nothing.

I would go with the following:
1. improve general "cloning" of operations in the `onChange` handler in this PR
2. handle edge cases like using the operations `addOperationHandler` in the fix for #51780

Is that fine for you @aocenas ?

Comment (aocenas) [reply public/app/plugins/datasource/prometheus/querybuilder/shared/OperationHeader.tsx]: Yeah sure, honestly maybe just checking type is ok but will defer that to you just wanted to ask if that was considered and we don't end up with having a label name in a field expecting number which I assume could also blow up somewhere.

**Pull request merged**